Mathlib/LinearAlgebra/Finsupp/LinearCombination.lean

```diff
@@ -306,16 +325,43 @@ theorem Finsupp.linearCombination_eq_fintype_linearCombination_apply (x : α →
 
 theorem Finsupp.linearCombination_eq_fintype_linearCombination :
     (linearCombination R v).comp (Finsupp.linearEquivFunOnFinite R R α).symm.toLinearMap =
-      Fintype.linearCombination R S v :=
-  LinearMap.ext <| linearCombination_eq_fintype_linearCombination_apply R S v
-
-variable {S}
+      Fintype.linearCombination R v :=
+  LinearMap.ext <| linearCombination_eq_fintype_linearCombination_apply R v
 
 @[simp]
 theorem Fintype.range_linearCombination :
-    LinearMap.range (Fintype.linearCombination R S v) = Submodule.span R (Set.range v) := by
+    LinearMap.range (Fintype.linearCombination R v) = Submodule.span R (Set.range v) := by
   rw [← Finsupp.linearCombination_eq_fintype_linearCombination, LinearMap.range_comp,
       LinearEquiv.range, Submodule.map_top, Finsupp.range_linearCombination]
+
+/-- `Fintype.bilinearCombination R S v f` is the linear combination of vectors in `v` with weights
+in `f`. This variant of `Finsupp.linearCombination` is defined on fintype indexed vectors.
+
+This map is linear in `v` if `R` is commutative, and always linear in `f`.
+See note [bundled maps over different rings] for why separate `R` and `S` semirings are used.
+-/
+protected def Fintype.bilinearCombination : (α → M) →ₗ[S] (α → R) →ₗ[R] M where
+  toFun v := Fintype.linearCombination R v
+  map_add' u v := by ext; simp [Fintype.linearCombination,
+    Finset.sum_add_distrib, Pi.add_apply, smul_add]
+  map_smul' r v := by ext; simp [Fintype.linearCombination, Finset.smul_sum, smul_comm]
+
+variable {S}
+
+theorem Fintype.bilinearCombination_apply_eq_linearCombination :
+    Fintype.bilinearCombination R S v = Fintype.linearCombination R v :=
+  rfl
+
+theorem Fintype.bilinearCombination_apply (f) :
+    Fintype.bilinearCombination R S v f = ∑ i, f i • v i :=
+  rfl
+
+@[simp]
+theorem Fintype.bilinearCombination_apply_single [DecidableEq α] (i : α) (r : R) :
+    Fintype.bilinearCombination R S v (Pi.single i r) = r • v i := by
+  simp_rw [Fintype.bilinearCombination_apply, Pi.single_apply, ite_smul, zero_smul]
+  rw [Finset.sum_ite_eq', if_pos (Finset.mem_univ _)]
```

Comment:
This should be proved about `linearCombination` instead (or dropped entirely if that already exists), since with the above the LHS is no longer simp-normal.

Reply:
`simp` proves it already, hence I deleted it.